File: app/src/main/java/com/commit451/gitlab/viewHolder/MilestoneHeaderViewHolder.java
package com.commit451.gitlab.viewHolder;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.commit451.gitlab.R;
import com.commit451.gitlab.model.api.Milestone;
import com.commit451.gitlab.util.DateUtil;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * Header with information for milestones
 */
public class MilestoneHeaderViewHolder extends RecyclerView.ViewHolder {

    public static MilestoneHeaderViewHolder inflate(ViewGroup parent) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.header_milestone, parent, false);
        return new MilestoneHeaderViewHolder(view);
    }

    @BindView(R.id.description)
    TextView textDescription;
    @BindView(R.id.due_date)
    TextView textDueDate;

    public MilestoneHeaderViewHolder(View view) {
        super(view);
        ButterKnife.bind(this, view);
    }

    public void bind(Milestone milestone) {
        if (milestone.getDescription() != null) {
            textDescription.setText(milestone.getDescription());
        }
        if (milestone.getDueDate() != null) {
            CharSequence due = DateUtil.getRelativeTimeSpanString(itemView.getContext(), milestone.getDueDate());
            textDueDate.setText(String.format(itemView.getResources().getString(R.string.due_date_formatted), due));
        }
    }
}

File: app/src/main/java/com/commit451/gitlab/viewHolder/MilestoneHeaderViewHolder.kt
package com.commit451.gitlab.viewHolder

import androidx.recyclerview.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import com.commit451.addendum.recyclerview.bindView
import com.commit451.gitlab.R
import com.commit451.gitlab.model.api.Milestone
import com.commit451.gitlab.util.DateUtil

/**
 * Header with information for milestones
 */
class MilestoneHeaderViewHolder(view: View) : RecyclerView.ViewHolder(view) {

    companion object {

        fun inflate(parent: ViewGroup): MilestoneHeaderViewHolder {
            val view = LayoutInflater.from(parent.context)
                    .inflate(R.layout.header_milestone, parent, false)
            return MilestoneHeaderViewHolder(view)
        }
    }

    private val textDescription: TextView by bindView(R.id.description)
    private val textDueDate: TextView by bindView(R.id.due_date)

    fun bind(milestone: Milestone) {
        if (milestone.description != null) {
            textDescription.text = milestone.description
        }
        if (milestone.dueDate != null) {
            val due = DateUtil.getRelativeTimeSpanString(itemView.context, milestone.dueDate!!)
            textDueDate.text = String.format(itemView.resources.getString(R.string.due_date_formatted), due)
        }
    }
}
